vortex-array/src/validity.rs

```diff
@@ -489,9 +490,12 @@ impl LogicalValidity {
         }
     }
 
-    pub fn into_validity(self) -> Validity {
+    pub fn into_validity(self, nullability: Nullability) -> Validity {
         match self {
```

Comment:
You should panic in here if nullability is NonNullable but logical is AllInvalid

Reply:
here you go. I was hoping we can do a stronger assertion, i.e. NonNullable implies AllValid but looks like in some cases we get an Array of all true :/